src/llama.cpp

```diff
@@ -5517,6 +5517,12 @@ static void llm_load_vocab(
                 tokenizer_pre == "viking") {
                 vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_VIKING;
                 vocab.tokenizer_clean_spaces = false;
+            } else if (
+                tokenizer_pre == "tekken") {
+                vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_TEKKEN;
+                vocab.tokenizer_clean_spaces = true;
+                vocab.tokenizer_ignore_merges = true;
+                vocab.tokenizer_add_bos = true;
```

Comment:
```suggestion
                vocab.tokenizer_ignore_merges = true;
                vocab.tokenizer_add_bos = true;
```
`tokenizer_clean_spaces` defaults to `true` for all BPE tokenizers (see line 5440), so no need to set it explicitly here.